File: src/main/java/com/dozycoffee/wms/global/error/ErrorTypeHttpStatusMapper.java
package com.dozycoffee.wms.global.error;

import org.springframework.http.HttpStatus;

public final class ErrorTypeHttpStatusMapper {

    private ErrorTypeHttpStatusMapper() {
    }

    public static HttpStatus resolve(ErrorType errorType) {
        return switch (errorType) {
            case VALIDATION -> HttpStatus.BAD_REQUEST;
            case NOT_FOUND -> HttpStatus.NOT_FOUND;
            case CONFLICT -> HttpStatus.CONFLICT;
            case INTERNAL -> HttpStatus.INTERNAL_SERVER_ERROR;
        };
    }
}

File: src/main/kotlin/com/dozycoffee/wms/global/error/ErrorTypeHttpStatusMapper.kt
package com.dozycoffee.wms.global.error

import org.springframework.http.HttpStatus

object ErrorTypeHttpStatusMapper {

    fun resolve(errorType: ErrorType): HttpStatus {
        return when (errorType) {
            ErrorType.VALIDATION -> HttpStatus.BAD_REQUEST
            ErrorType.NOT_FOUND -> HttpStatus.NOT_FOUND
            ErrorType.CONFLICT -> HttpStatus.CONFLICT
            ErrorType.INTERNAL -> HttpStatus.INTERNAL_SERVER_ERROR
        }
    }
}
